java/code/src/com/redhat/rhn/taskomatic/task/ForwardRegistrationTask.java

```diff
@@ -94,19 +92,51 @@ public void execute(JobExecutionContext arg0) {
                 waitTime = 1;
             }
             try {
-                Thread.sleep(waitTime * 1000);
+                Thread.sleep(waitTime * 1000L);
             }
             catch (InterruptedException e) {
                 log.debug("Sleep interrupted", e);
+                Thread.currentThread().interrupt();
             }
-            optPrimCred.ifPresent(primaryCredentials -> executeSCCTasks(primaryCredentials));
+            optPrimCred.ifPresent(credentials -> executeSCCTasks(credentials));
         }
     }
 
-    /*
-     * Do SCC related tasks like insert, update and delete system in SCC
-     */
-    private void executeSCCTasks(SCCCredentials primaryCredentials) {
+    protected Optional<SCCCredentials> findSccCredentials() {
+        HubFactory hubFactory = new HubFactory();
+        Optional<IssHub> optHub = hubFactory.lookupIssHub();
+        return optHub.map(this::getSCCCredentialsWhenHub).orElse(getSCCCredentials());
+    }
+
+    protected Optional<SCCCredentials> getSCCCredentials() {
+        List<SCCCredentials> credentials = CredentialsFactory.listSCCCredentials();
+        Optional<SCCCredentials> optPrimCred = credentials.stream()
+                .filter(c -> c.isPrimary())
+                .findFirst();
+        if (optPrimCred.isEmpty()) {
+            // We cannot update SCC without credentials
+            // Standard Uyuni case
+            log.debug("No SCC Credentials - skipping forwarding registration");
+        }
+
+        return optPrimCred;
+    }
+
+    protected Optional<SCCCredentials> getSCCCredentialsWhenHub(IssHub hub) {
+        List<SCCCredentials> credentials = CredentialsFactory.listSCCCredentials();
+        Optional<SCCCredentials> optHubCredential = credentials.stream()
+                .filter(c -> hub.getFqdn().equals(c.getUrl()))
```

Comment:
I think this will never match. You compare a FQDN with a URL. The url has always the "https://" in front of the FQDN which make the equals fail.

As you have the hub object, get the credentials from there.

```
 Optional<SCCCredentials> optHubCredential = Optional.ofNullable(hub.getMirrorCredentials());
```